src/Doctrine/Annotation/DocLexer.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Doctrine\Annotation;
+
+use PhpCsFixer\Preg;
+
+/**
+ * Copyright (c) 2006-2013 Doctrine Project.
```

Comment:
only as an option - not as must have

if easier, we could put the "copied" code to https://github.com/php-cs-fixer/doctrine etc.

we done sth like this in past already https://github.com/PHP-CS-Fixer/diff


if overkill, feel free to ignore the comment

Reply:
I feel like it's an overkill because we would need to set up the CI process somehow, while currently all the Doctrine-related fixers are included in the test suite here. This change solves the root problem (superfluous required dependencies) with minimal cost of maintaining additional lexer class. If anything, such extraction could be done in separate PR later, so let's merge it as-is.